drivers/syslog/syslog_filechannel.c

```diff
@@ -50,6 +54,62 @@
 
 FAR static struct syslog_channel_s *g_syslog_file_channel;
 
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+#ifdef CONFIG_SYSLOG_FILE_ROTATE
+static void log_rotate(FAR const char *log_file)
+{
+  int fd;
+  off_t size;
+  struct stat f_stat;
+  char *backup_file;
+
+  /* Get the size of the current log file. */
+
+  fd = open(log_file, O_RDONLY);
```

Comment:
it's prefer to call nx_xxx or file_xxx in kernel space.

Reply:
Fixed in #3848.